api/tests/unit/users/test_unit_users_models.py

```diff
@@ -1,168 +1,154 @@
-            name="Test project 1", organisation=self.organisation
-        )
-        self.project_2 = Project.objects.create(
-            name="Test project 2", organisation=self.organisation
-        )
 
-        self.environment_1 = Environment.objects.create(
-            name="Test Environment 1", project=self.project_1
-        )
-        self.environment_2 = Environment.objects.create(
-            name="Test Environment 2", project=self.project_2
-        )
+def test_user_belongs_to_fail(admin_user: FFAdminUser) -> None:
+    unaffiliated_organisation = Organisation.objects.create(name="Unaffiliated")
+    assert not admin_user.belongs_to(unaffiliated_organisation.id)
 
-    def test_user_belongs_to_success(self):
-        self.user.add_organisation(self.organisation, OrganisationRole.USER)
-        assert self.user.belongs_to(self.organisation.id)
 
-    def test_user_belongs_to_fail(self):
-        assert not self.user.belongs_to(self.organisation.id)
+def test_get_permitted_projects_for_org_admin_returns_all_projects(
+    admin_user: FFAdminUser,
+    organisation: Organisation,
+    project: Project,
+) -> None:
+    # Given
+    Project.objects.create(name="Test project 2", organisation=organisation)
+    # When
+    projects = admin_user.get_permitted_projects(VIEW_PROJECT)
 
-    def test_get_permitted_projects_for_org_admin_returns_all_projects(self):
-        # Given
-        self.user.add_organisation(self.organisation, OrganisationRole.ADMIN)
+    # Then
+    assert projects.count() == 2
 
-        # When
-        projects = self.user.get_permitted_projects(VIEW_PROJECT)
 
-        # Then
-        assert projects.count() == 2
+def test_get_permitted_projects_for_user_returns_only_projects_matching_permission(
+    staff_user: FFAdminUser,
+    with_project_permissions: WithProjectPermissionsCallable,
+    project: Project,
+) -> None:
+    # Given
+    with_project_permissions([VIEW_PROJECT])
 
-    def test_get_permitted_projects_for_user_returns_only_projects_matching_permission(
-        self,
-    ):
-        # Given
-        self.user.add_organisation(self.organisation, OrganisationRole.USER)
-        user_project_permission = UserProjectPermission.objects.create(
-            user=self.user, project=self.project_1
-        )
-        read_permission = ProjectPermissionModel.objects.get(key=VIEW_PROJECT)
-        user_project_permission.permissions.set([read_permission])
+    # When
+    projects = staff_user.get_permitted_projects(permission_key=VIEW_PROJECT)
 
-        # When
-        projects = self.user.get_permitted_projects(permission_key=VIEW_PROJECT)
+    # Then
+    assert projects.count() == 1
+    assert projects.first() == project
 
-        # Then
-        assert projects.count() == 1
 
-    def test_get_admin_organisations(self):
-        # Given
-        self.user.add_organisation(self.organisation, OrganisationRole.ADMIN)
+def test_get_admin_organisations(
```

Comment:
Can we create another organization here and assign the user as a non-administrator to it, just to make sure it's not returning all the organizations

Reply:
Yes this is a good idea. I should have caught that with the existing test.

Reply:
Ok this is fixed. I also found a typing issue in another file that was missed by a recent cleanse so I just added it into this PR.